ironfist/src/cpp/combat/creatures.cpp

```diff
@@ -52,16 +54,9 @@ attributeNameTableEntry creatureAttributeNameTable[] = {
 	{"undead", UNDEAD}
 };
 
-struct secondaryCostIdxTableEntry { char * resource; int idx; };
 
-secondaryCostIdxTableEntry secondaryCostIdxTable[] = {
-	{"wood", 0},
-	{"mercury", 1},
-	{"ore", 2},
-	{"sulfer", 3},
-	{"crystal", 4},
-	{"gems", 5}
-};
+string resourceName[] = { "wood", "mercury", "ore", "sulfur", "crystal", "gems", "gold" };

```

Comment:
This is dangerous. You're no counting on the order declared here to be the same as the one given in another file. I think I already suggested a decent way of doing this in my previous comments.